app/src/main/java/com/example/event_finder_capstone/PhotoAlbumActivity.java

```diff
@@ -0,0 +1,94 @@
+package com.example.event_finder_capstone;
+
+import android.content.Intent;
+import android.os.Bundle;
+import android.view.Menu;
+import android.view.MenuItem;
+import android.widget.Toast;
+
+import androidx.annotation.Nullable;
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.recyclerview.widget.GridLayoutManager;
+import androidx.recyclerview.widget.RecyclerView;
+
+import com.example.event_finder_capstone.models.Event;
+import com.example.event_finder_capstone.models.Photo;
+import com.parse.FindCallback;
+import com.parse.ParseException;
+import com.parse.ParseQuery;
+
+import org.parceler.Parcels;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class PhotoAlbumActivity extends AppCompatActivity {
+
+    private static final String TAG = "PhotoAlbumActivity";
+    private static final int POST_LIMIT = 20;
+    protected PhotoAdapter photo_adapter;
+    protected List<Photo> allPhotos;
+    Event event;
+    RecyclerView rvPhotos;
+
+    @Override
+    protected void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_photo_album);
+
+        event = Parcels.unwrap(getIntent().getParcelableExtra(Event.class.getSimpleName()));
+        getSupportActionBar().setTitle("Photos from " + event.getName());
+
+        rvPhotos = findViewById(R.id.rvPhotos);
```

Comment:
⚠ Note:  Make sure you're working toward consistently breaking larger blocks of code into separate methods.  In this case, I think there is an opportunity to create a method along the lines of `seutpPhotoGridRecyclerView()`